### ISSUE TITLE
Clean setup instruction doesn't work at the end(02/05/18). Process stuck in the Backfilling

### ISSUE THREAD

**Issue by jeeseop** [DeviaVir/zenbot#1290]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/zenbot

If you open a GitHub issue, here is our policy:

1. It must be a bug or a feature request.
2. The form below must be filled out.

**Here's why we have that policy**: Zenbot developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies hundreds of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using zenbot vanilla)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: 16.04
- **Zenbot version**: current one
- **NodeJS version**: 8.9.4
- **Python version**: 2.7.12
- **Exact command to reproduce**:

### Describe the problem

I installed ubuntu from the beginning and I followed the instruction from the github and blog post both of them. But both of them stops at the last command: ./zenbot.sh trade --paper

It stops in here: 

fetching pre-roll data:
*************************
Hitting up the exchange 'gdax' for trades within the past 1 day.
*************************

Backfilling...

How can solve this problem?

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by DeviaVir** [comment]
How long did you wait before filing this ticket?

**Comment by jeeseop** [comment]
@DeviaVir I waited more than 3 hours.

Is it still short?

**Comment by jeeseop** [comment]
I changed several exchanges,

@DeviaVir I can see that the binance and gdax doesn't work anytime.
And Poloniex data is sometimes received.

I spent whole day for testing this.
I can give you my word that I spend at least one hours for choosen exchange currency pair

**Comment by DeviaVir** [comment]
I know we have issues with some exchanges doing backfilling: https://github.com/DeviaVir/zenbot/issues

However GDAX and Binance were the ones I thought were working, so that is surprising.

**Comment by jeeseop** [comment]
@DeviaVir Firstly, I thought that there is a problem in my computer and I tried to fix it.
I reinstall ubuntu several times and tried to install zenbot several times, and I got a result that some exchange connection doesn't work.
I'm so sorry for that I couldn't give you a suspicious point of this situation.

**Comment by jeeseop** [comment]
@DeviaVir 
I firstly thought that if some branch merging can be a problem, but I don't have enough ability to check all of the previous snapshots.

**Comment by station384** [comment]
TL;DR Try updating to latest unstable

I don't see if your referencing release,  or the unstable branch.  If it is from the unstable branch this may explain what is going on.
Pull #1214 broke the backfilling for all exchanges, it didn't include the field selector when inserting new records into trade.db, it was retrieving the back data,  but sim and trade couldn't see them as they use the field selector when finding those records.   

This has since been corrected in pull #1279.

As of 10hrs ago in unstable it was working correctly.  I tested this by dropping the zenbot database in mongo db, and starting over using backfilling.
I have just synced up with the latest pulls and everything still seems to  be working with bittrex exchange atleast, I have gdax currently trading but I have not tested backfilling specifically ( I paid little attention if it got everything or not, but it started, and pulled records from the past what would not have been there if backfilling did not work, so I assume it worked )

**Comment by jeeseop** [comment]
@station384 
I downloaded yesterday 10 am version zip file too. And it also had a problem above.
Is this version can be a #1214?

**Comment by station384** [comment]
Github isn't displaying a specific time when it was merged.   it is just saying a day ago.  I would suggest updating and checking (make sure you backup conf.js so you don't have to reenter that stuff ).
gdax is also very slow when backfilling.   the more --days you pick the longer it will take.   I usually just do --day 1 for testing to make sure its working, and it still takes a long time to fill.  blame gdax for this,   zen just requests then gdax spoon feeds the results at its own pace,  this pace can be at times very fast when there load is low, but others it takes an exceptionally long time I've noticed.

**Comment by jeeseop** [comment]
@DeviaVir  @station384 

Now the binance trading is working.
gdax api sometimes down but I'm not sure which side is wrong.

Where can I see the code upgrade story for backfilling stuck problem at  this time?

**Comment by DeviaVir** [comment]
Is this still an active issue today, on latest `unstable`?
